src/main/java/org/apache/lucene/analysis/kr/utils/DictionaryUtil.java

```diff
@@ -18,36 +18,127 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import org.apache.lucene.analysis.kr.morph.CompoundEntry;
 import org.apache.lucene.analysis.kr.morph.MorphException;
 import org.apache.lucene.analysis.kr.morph.WordEntry;
 
 public class DictionaryUtil {
 	
-	private static Trie<String,WordEntry> dictionary;
+	private static Supplier<Trie<String,WordEntry>> DICTIONARY = Suppliers.memoize(new Supplier<Trie<String, WordEntry>>() {
+        @Override
+        public Trie<String, WordEntry> get() {
+            try {
+                return loadDictionary();
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize DICTIONARY.", e);
+            }
+        }
+    });
 	
-	private static HashMap josas;
+	private static Supplier<Map<String, String>> JOSAS = Suppliers.memoize(new Supplier<Map<String, String>>() {
+        @Override
+        public Map<String, String> get() {
+            try {
+                return readDict(KoreanEnv.FILE_JOSA);
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize JOSA dictionary.", e);
+            }
+        }
+    });
 	
-	private static HashMap eomis;
-	
-	private static HashMap prefixs;
-	
-	private static HashMap suffixs;
-	
-	private static HashMap<String,WordEntry> uncompounds;
+	private static Supplier<Map<String, String>> EOMIS = Suppliers.memoize(new Supplier<Map<String, String>>() {
+        @Override
+        public Map<String, String> get() {
+            try {
+                return readDict(KoreanEnv.FILE_EOMI);
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize EOMI dictionary.", e);
+            }
+        }
+    });
+
+    private static Supplier<Map<String, String>> PREFIXS = Suppliers.memoize(new Supplier<Map<String, String>>() {
+        @Override
+        public Map<String, String> get() {
+            try {
+                return readDict(KoreanEnv.FILE_PREFIX);
+            } catch (MorphException e) {
```

Comment:
seems like all the try/catch layout is repetitive and needs to be factored out into its own method?